app/models/imports/podcast_rss_import.rb

```diff
@@ -186,32 +185,27 @@ def create_or_update_podcast!
     podcast
   end
 
-  def enclosure_prefix(podcast_item)
-    prefix = ""
-    link = [podcast_item.feedburner_orig_enclosure_link,
-      podcast_item.enclosure.try(:url),
-      podcast_item.media_contents.first.try(:url)].find do |url|
-      url.try(:match, /podtrac/) || url.try(:match, /blubrry/)
-    end
-    if (scheme = link.try(:match, /^https?:\/\//))
-      prefix += scheme.to_s
-    end
-    if (podtrac = link.try(:match, /\/(\w+\.podtrac.com\/.+?\.mp3\/)/))
-      prefix += podtrac[1]
-    end
-    if (blubrry = link.try(:match, /\/(media\.blubrry\.com\/[^\/]+\/)/))
-      prefix += blubrry[1]
-    end
-    prefix
-  end
+  def enclosure_prefix(item)
+    redirectors = [
+      /\/(www|dts)\.podtrac\.com(\/pts)?\/redirect\.mp3\//,
+      /\/media\.blubrry\.com\/[^\/]+\//,
+      /\/chrt\.fm\/track\/[^\/]+\//,
+      /\/chtbl\.com\/track\/[^\/]+\//,
+      /\/pdst\.fm\/e\//
+    ]
+
+    urls = [item.feedburner_orig_enclosure_link, item.enclosure.try(:url), item.media_contents.first.try(:url)]
+    url = urls.compact.find { |u| redirectors.any? { |r| u.match?(r) } }
 
-  def feedburner_url(fb_name)
-    fb_name ? "https://feeds.feedburner.com/#{clean_string(fb_name)}" : nil
+    if url.present?
+      end_index = redirectors.map { |r| url.index(r) + url[r].length if url.match?(r) }.compact.max
+      url[0...end_index]
```

Comment:
Refactored to support more redirect prefixes.

This code (1) looks for an enclosure/contents/etc url with any redirect prefix in it.  And then (2) returns the entire redirect chain (in case multiple), minus the final Mp3-hosting domain.

Reply:
if i'm reading correctly, (1) is handled by L198, and (2) is L201-202? is the `url` mutated on L202 with that range method?

Reply:
yep, exactly